lib/prepareFunctionsUpload.js

```diff
@@ -92,7 +92,11 @@ var _packageSource = function(options, tmpdir) {
 
     reader.on('end', function() {
       archive.finalize();
-      fs.removeSync(tmpdir.name);
+      try {
+        fs.removeSync(tmpdir.name);
+      } catch (e) {
+        logger.debug('Cannot remove ', tmpdir.name, e);
```

Comment:
Let's use `utils.logWarning('Unable to delete temporary directory ' + tmpdir.name)` here. The user should be aware that there's potentially a bunch of extra disk space being left uncleaned.